core/metadata.js

```diff
@@ -178,6 +179,12 @@ function ensureValidPath (doc /*: PathObject */) {
   }
 }
 
+function invariants (doc /*: Metadata */) {
```

Comment:
Rename to `validateBeforeSave` or `validateForSaving`, since `Metadata` can still exist without `sides` before `Prep`, the `invariants` name could be confusing?